packages/build/src/plugins/compatibility.ts

```diff
@@ -81,29 +80,37 @@ const getCompatibleEntry = async function ({
   packagePath?: string
   pinnedVersion?: string
 }): Promise<Pick<PluginVersion, 'conditions' | 'version'>> {
-  const compatibleEntry = await pLocate(versions, async ({ version, overridePinnedVersion, conditions }) => {
-    // Detect if the overridePinnedVersion intersects with the pinned version in this case we don't care about filtering
-    const overridesPin = Boolean(
-      pinnedVersion && overridePinnedVersion && semver.intersects(overridePinnedVersion, pinnedVersion),
-    )
+  const versionsWithCheckConditions = await Promise.all(
+    versions.map(async (versionInfo) => {
+      const { version, overridePinnedVersion, conditions } = versionInfo
+      // Detect if the overridePinnedVersion intersects with the pinned version in this case we don't care about filtering
+      const overridesPin = Boolean(
+        pinnedVersion && overridePinnedVersion && semver.intersects(overridePinnedVersion, pinnedVersion),
+      )
 
-    // ignore versions that don't satisfy the pinned version here if a pinned version is set
-    if (!overridesPin && pinnedVersion && !semver.satisfies(version, pinnedVersion, { includePrerelease: true })) {
-      return false
-    }
+      // ignore versions that don't satisfy the pinned version here if a pinned version is set
+      if (!overridesPin && pinnedVersion && !semver.satisfies(version, pinnedVersion, { includePrerelease: true })) {
+        return { versionInfo, compatible: false, constraintsScore: 0 }
+      }
 
-    // no conditions means nothing to filter
-    if (conditions.length === 0 && pinnedVersion === undefined) {
-      return false
-    }
+      // no conditions means nothing to filter
+      if (conditions.length === 0 && pinnedVersion === undefined) {
+        return { versionInfo, compatible: true, constraintsScore: 0 }
```

Comment:
this marks this is compatible instead of ignoring it previously - it assign lowest score so it's sorted out as last later in case there were compatible ones with constraints